components/command/runner.go

```diff
@@ -21,48 +22,113 @@ type Args struct {
 	Environment              pulumi.StringMap
 	RequirePasswordFromStdin bool
 	Sudo                     bool
+	// Only used for local commands
+	LocalAssetPaths pulumi.StringArrayInput
+	LocalDir        pulumi.StringInput
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
For this, you could have a `LocalArguments` interface, that only a new struct `LocalArgs` implement, and cast `Args` to `LocalArgs` here, failing if the cast fails.

This would be something like

```go
type Args struct {
	Create                   pulumi.StringInput
	Update                   pulumi.StringInput
	Delete                   pulumi.StringInput
	Triggers                 pulumi.ArrayInput
	Stdin                    pulumi.StringPtrInput
	Environment              pulumi.StringMap
	RequirePasswordFromStdin bool
	Sudo                     bool
}

type LocalArguments interface {
	LocalAssetPaths() pulumi.StringArrayInput
	LocalDir() 		       pulumi.StringInput
}

type LocalArgs struct {
	Args
	LocalAssetPaths pulumi.StringArrayInput
	LocalDir        pulumi.StringInput  
}

var _ LocalArgument = *LocalArgs{}

func (args *LocalArgs) LocalAssetPaths() pulumi.StringArrayInput {
	return args.LocalAssetPaths
}

func (args *LocalArgs) LocalDir() pulumi.StringInput {
	return args.LocalDir
}

func (args *Args) toLocalCommandArgs(config RunnerConfiguration, osCommand OSCommand) (*local.CommandArgs, error) {
	if localArgs, ok := args.(LocalArgs); !ok {
		return nil, fmt.Errorf("can convert only local args")
	} 
	return &local.CommandArgs{
		Create:   osCommand.Build
		Create:      osCommand.BuildCommandString(args.Create, args.Environment, args.Sudo, args.RequirePasswordFromStdin, config.user),
		Update:      osCommand.BuildCommandString(args.Update, args.Environment, args.Sudo, args.RequirePasswordFromStdin, config.user),
		Delete:      osCommand.BuildCommandString(args.Delete, args.Environment, args.Sudo, args.RequirePasswordFromStdin, config.user),
		Environment: args.Environment,
		Triggers:    args.Triggers,
		Stdin:       args.Stdin,
		AssetPaths:  args.LocalAssetPaths,
		Dir:         args.LocalDir,
	}, nil
}
```

Don't know if you can possibly define `toLocalCommandArgs` on `LocalArgs` rather than on `Args`, I think you showed me yesterday you could not